### PR TITLE
Better --version handling

### PR DESCRIPTION
**Pull request by Jorres**
Before this PR, `--version ==MAJOR.MINOR.PATCH` matched both `23.3.1-ydb-stable-hotfix-1` and `23.3.1-ydb-stable-hotfix-2`. It WAS intended behaviour, utility looked at first `MAJOR.MINOR.PATCH` fragment in the node version.

@mvgorbunov mentioned it could lead to an incident, restarting something that is not supposed to be restarted. So the behaviour was split into two:

`--version ==STRING_LITERAL` now allows matching for the whole version string (e.g. you can now restart `--version ==23.3.1-ydb-stable-hotfix-1` and `...hotfix-2` will not be selected. 

`--version ~=MAJOR.MINOR.PATCH` behaves just like `--version ==MAJOR.MINOR.PATCH` behaved before this PR. 

To summarize:

1. It is now possible to match the entire version string, even if it's super weird like `stream-nb-tracing.1.<commit-hash>`
2. Changing == to ~=, it is a bit more explicit to the user that he selects a wideR configuration and needs to pay more attention.

### CODE REVIEW

File: pkg/rolling/options.go [hunk 1]
```diff
@@ -253,3 +244,38 @@ func (o *RestartOptions) GetNodeIds() ([]uint32, error) {
 
 	return ids, nil
 }
+
+func parseVersionFlag(versionUnparsedFlag string) (options.VersionSpec, error) {
+	majorMinorPatchPattern := `^(>|<|!=|~=)(\d+|\*)\.(\d+|\*)\.(\d+|\*)$`
+	re1 := regexp.MustCompile(majorMinorPatchPattern)
```
Comment (shmel1k) [inline pkg/rolling/options.go:250]: It's highly recommended to move MustCompile regexps out of functions to global variables: saves you a lot of time in loaded applications

File: tests/rolling_test.go [hunk 2]
```diff
@@ -379,6 +379,68 @@ var _ = Describe("Test Rolling", func() {
...
+				{
+					ydbopsInvocation: []string{
+						"--endpoint", "grpcs://localhost:2135",
+						"--verbose",
+						"--availability-mode", "strong",
+						"--user", mock.TestUser,
+						"--cms-query-interval", "1",
+						"run",
+						"--storage",
+						"--version", "==full_version_string_1234",
```
Comment (shmel1k) [inline tests/rolling_test.go:407]: Looks good to add some "table" tests for that.

For example, func TestParseSomething(t *testing.T) {}

**Review by shmel1k — CHANGES_REQUESTED** [review]

**Review by shmel1k — APPROVED** [review]
